packages/@aws-cdk/aws-ecs/lib/task-definition.ts

```diff
@@ -0,0 +1,260 @@
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation } from './ecs.generated';
+
+export interface TaskDefinitionProps {
+
+    containerDefinitions: ContainerDefinition[];
+
+    /**
+     * The number of cpu units used by the task. If using the EC2 launch type,
+     * this field is optional. Supported values are between 128 CPU units
+     * (0.125 vCPUs) and 10240 CPU units (10 vCPUs). If you are using the
+     * Fargate launch type, this field is required and you must use one of the
+     * following values, which determines your range of valid values for the
+     * memory parameter:
+     * 256 (.25 vCPU) - Available memory values: 0.5GB, 1GB, 2GB
+     * 512 (.5 vCPU) - Available memory values: 1GB, 2GB, 3GB, 4GB
+     * 1024 (1 vCPU) - Available memory values: 2GB, 3GB, 4GB, 5GB, 6GB, 7GB, 8GB
+     * 2048 (2 vCPU) - Available memory values: Between 4GB and 16GB in 1GB increments
+     * 4096 (4 vCPU) - Available memory values: Between 8GB and 30GB in 1GB increments
+     *
+     * @default 256
+     */
+    cpu?: string;
+
+    /**
+     * The Amazon Resource Name (ARN) of the task execution role that
+     * containers in this task can assume. All containers in this task are
+     * granted the permissions that are specified in this role.
+     *
+     * Needed in Fargate to communicate with Cloudwatch Logs and ECR.
+     */
+    executionRoleArn?: string;
+
+    /**
+     * Namespace for task definition versions
+     *
+     * @default CloudFormation-generated name
```

Comment:
Yes but that behavior is not the same as with other CloudFormation-generated names, is it? It's a new name on every deployment?

Since that's the case, I'd rather use `this.uniqueId` as the default.